core/blockchain.go

```diff
@@ -2373,6 +2380,19 @@ func (bc *BlockChain) UpdateValidatorVotingPower(
 					}
 				} else {
 					stats.APR = *aprComputed
```

Comment:
why not just add the history APRs into the existing stats structure?

Reply:
stats are overwritten. no snapshotting. didn't want to snapshot the whole stats, hence separately snapshot the APRs. any other alternatives?

Reply:
add like a slice or map inside the stats?

Reply:
changed to fixed length circular slice inside stats.